cmd/gardenlet/app/migration.go

```diff
@@ -149,3 +163,38 @@ func createOSCHashMigrationSecret(ctx context.Context, seedClient client.Client)
 	}
 	return flow.Parallel(tasks...)(ctx)
 }
+
+// TODO (shafeeqes): Remove this function in gardener v1.100
+func cleanupDWDAccess(ctx context.Context, gardenClient client.Client, seedClient client.Client, seedName string) error {
+	shootList := &gardencorev1beta1.ShootList{}
+	if err := gardenClient.List(ctx, shootList, client.MatchingFields{core.ShootSeedName: seedName}); err != nil {
+		return err
+	}
+
+	workerlessShootNamespaces := sets.New[string]()
+	for _, shoot := range shootList.Items {
+		if v1beta1helper.IsWorkerless(&shoot) && shoot.DeletionTimestamp == nil {
+			workerlessShootNamespaces.Insert(shoot.Status.TechnicalID)
+		}
+	}
+
+	var taskFns []flow.TaskFn
+
+	for _, namespace := range workerlessShootNamespaces.UnsortedList() {
+		taskFns = append(taskFns, func(ctx context.Context) error {
+			if err := kubernetesutils.DeleteObjects(ctx, seedClient,
+				&corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: dependencywatchdog.KubeConfigSecretName, Namespace: namespace}},
+			); err != nil {
+				return fmt.Errorf("failed to delete DWD access secret for namespace %q: %w", namespace, err)
+			}
+
+			if err := managedresources.DeleteForShoot(ctx, seedClient, namespace, dependencywatchdog.ManagedResourceName); err != nil {
+				return fmt.Errorf("failed to delete DWD managed resource for namespace %q: %w", namespace, err)
+			}
+
+			return nil
+		})
+	}
+
+	return flow.Parallel(taskFns...)(ctx)
+}
```

Comment:
nit: we don't need two for loops now, and can combine them into one
But overall looks good to me